### PR TITLE
Finished implementation for ProjectInfoFinder .

### PR DESCRIPTION
**Pull request by ClintonCao**
I also did the following things: 
- I've added `hashCode` method for all classes that overrides the `equals` method.
- The objects of  `WarningGrouper` and `Summarizers` are now also immutable.

P.S. There were build failures in the beginning, because for some reason Travis couldn't find the files when they are in the `Runnables` folder. This was fixed by moving the files into `src/test/resources/TestCode`

Please review and lemme know if there are any questions.

### CODE REVIEW

**Comment by ClintonCao** [comment]
Each interface is moved to the same package that has the class(es) that implement the interface.

**Comment by boninggong** [comment]
Lookgs good Clinton! Good job

**Comment by ClintonCao** [comment]
Fixed the comment, as requested by Sunwei.

File: src/main/java/BlueTurtle/finders/ProjectInfoFinder.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	 * 
+	 * @param srcDir
+	 *            the source directory to search in.
+	 */
+	public void findFiles(File srcDir) {
+		
+		// find all subdirectories;
+		File[] subdirs = srcDir.listFiles();
+		
+		// Go through all subdirectories
```
Comment (SunweiWang) [inline src/main/java/BlueTurtle/finders/ProjectInfoFinder.java]: Here miss 1 dot, and maybe sub-directories instead of subdirectories.

**Comment by SunweiWang** [comment]
Clinton fixed the problem as I requested. I will merge this.

**Pull request merged**